tutorials/nservicebus-101/lesson-2/tutorial.md

```diff
@@ -0,0 +1,191 @@
+
+A **system**, on the other hand, can be made up of multiple executables on multiple machines, and usually has multiple sources of information. A system must deal with connectivity constantly. A system is not an application, and furthermore, each executable within a system is not an application either. All must deal with connectivity on a constant basis.
+
+This is why an NServiceBus system will send messages, so that individual executables can communicate with each other in a reliable way. If there's a connectivity problem between executables, messages store the communication in a reliable way so that the executable on the receiving end can receive the message when it is able to do so.
+
+
+## Objectives
+
+By the end of this lesson, you will have learned:
+
+* How to define messages
+* How to define message handlers
+* How to send and receive a message locally
+* How to use NServiceBus's logging capabilities
+
+
+## What is a message?
+
+A **message** is a collection of data sent via one-way communication between two endpoints. In NServiceBus, we can define a message via a simple class.
+
+In this lesson, we'll focus on the simpler type of message: commands. In [Lesson 4: Publishing events](../lesson-4/) we'll expand to look at events as well.
+
+Defining a command is pretty easy. We just create a class and mark it with the `ICommand` marker interface.
+
+snippet:Command
+
+The marker interface has no implementation, but lets NServiceBus know that the class is a command so that it can build up some metadata about the message type when the endpoint starts up. Any properties you create within the message constitute the message data.
+
+The name of the command class is also important. A command is an order to do something, so it should be named in the [imperative tense](https://en.wikipedia.org/wiki/Imperative_mood). `PlaceOrder`, `UpdateCustomerStatus`, and `ChargeCreditCard` are all great names for commands, because they are phrased as a command and are very specific. You can easily guess exactly what each of those messages will do. `UserMessage`, on the other hand, is not a good example. It is not in the imperative, and it's not very specific either. Another developer should know exactly what a command's purpose is just by reading the name.
+
+When sending a message, the endpoint's message serializer will serialize an instance of the `DoSomething` class (let's say to JSON) and add that to the contents of the outgoing message that goes to the queue. On the other end, the receiving endpoint will deserialize the message from JSON back to an instance of the message class so that it can be used in code.
```

Comment:
remove "(let's say to JSON) " is adds no value. and instead make "message serializer" a link